columnar/src/writer/mod.rs

```diff
@@ -0,0 +1,528 @@
+mod column_operation;
+mod column_writers;
+mod serializer;
+mod value_index;
+
+use std::io::{self, Write};
+
+use column_operation::ColumnOperation;
+use fastfield_codecs::serialize::ValueIndexInfo;
+use fastfield_codecs::{Column, MonotonicallyMappableToU64, VecColumn};
+use serializer::ColumnarSerializer;
+use stacker::{Addr, ArenaHashMap, MemoryArena};
+
+use crate::column_type_header::{ColumnType, ColumnTypeAndCardinality, ColumnTypeCategory};
+use crate::dictionary::{DictionaryBuilder, TermIdMapping, UnorderedId};
+use crate::value::{Coerce, NumericalType, NumericalValue};
+use crate::writer::column_writers::{ColumnWriter, NumericalColumnWriter, StrColumnWriter};
+use crate::writer::value_index::{IndexBuilder, SpareIndexBuilders};
+use crate::{Cardinality, DocId};
+
+/// This is a set of buffers that are only here
+/// to limit the amount of allocation.
+#[derive(Default)]
+struct SpareBuffers {
+    value_index_builders: SpareIndexBuilders,
+    i64_values: Vec<i64>,
```

Comment:
we can drop the value buffers if we implement `Column` over the operations instead

Reply:
The operation are written sequentially, so that new doc op are interleaved with new value op.
That would mean doing several pass and ignore some data... Or that would mean writing into two different places.

I suspect the code is simpler and more efficient this way.